node/manager.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package node
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/iotexproject/go-pkgs/crypto"
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-core/pkg/log"
+	"github.com/iotexproject/iotex-core/pkg/routine"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+	"github.com/iotexproject/iotex-core/pkg/version"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type (
+	transmitter interface {
+		BroadcastOutbound(context.Context, proto.Message) error
+		UnicastOutbound(context.Context, peer.AddrInfo, proto.Message) error
+	}
+
+	heightable interface {
+		TipHeight() uint64
+	}
+
+	privateKey interface {
+		PublicKey() crypto.PublicKey
+		Sign([]byte) ([]byte, error)
+	}
+
+	// Info node infomation
+	Info struct {
+		Version   string
+		Height    uint64
+		Timestamp time.Time
+		Address   string
+	}
+
+	// DelegateManager manage delegate node info
+	DelegateManager struct {
+		cfg          Config
+		nodeMap      map[string]Info
+		nodeMapMutex sync.Mutex
+		broadcaster  *routine.RecurringTask
+		transmitter  transmitter
+		heightable   heightable
+		privKey      privateKey
+	}
+)
+
+// NewDelegateManager new delegate manager
+func NewDelegateManager(cfg *Config, t transmitter, h heightable, privKey privateKey) *DelegateManager {
+	dm := &DelegateManager{
+		cfg:         *cfg,
+		nodeMap:     make(map[string]Info),
+		transmitter: t,
+		heightable:  h,
+		privKey:     privKey,
+	}
+
+	dm.broadcaster = routine.NewRecurringTask(func() {
+		// delegates and nodes who are turned on will broadcast
+		if !dm.isDelegate() && dm.cfg.OnlyDelegateBroadcast {
+			log.L().Debug("delegate manager general node disabled node info broadcast")
+			return
+		}
```

Comment:
move this to before L72, if info broadcast is disabled, can early exit, no need to start the `NewRecurringTask`

Reply:
remain the task has an advantage that a new delegate can enable broadcast automatically without restart